core/market/src/negotiation/common.rs

```diff
@@ -333,6 +336,23 @@ impl CommonBroker {
             &agreement.id,
             reason.display(),
         );
+
+        // If we are Requestor we would like to be able to negotiate with the same
+        // provider for the second time, so we must generate new Proposal for him.
+        // Note: Regeneration failure isn't propagated to Requestor, because termination
+        // succeeded at this point.
+        if let OwnerType::Requestor = agreement.id.owner() {
+            self.regenerate_proposal(&agreement)
+                .await
+                .map_err(|e| {
+                    log::warn!(
+                        "Failed to regenerate Proposal after Agreement [{}] termination. {}",
+                        &agreement.id,
+                        e
+                    )
+                })
```

Comment:
this logic should be enclosed within `regenerate_proposal`

Reply:
It wouldn't be elegant. Check `regenerate_proposal` internals. I would have to add this warning in 3 places or create wrapper function that does only this

Reply:
ok, let it be